File: demokrat_back/demokrat_back/Classes/Class.cs
using demokrat_back.Users;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace demokrat_back.Classes
{
    public class Class
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public DateTime DateTime { get; set; }
        [Required]
        [MaxLength(200)]
        public string Location { get; set; } = string.Empty;
        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal Price { get; set; }
        [Required]
        public Guid TrainerId { get; set; }
        [ForeignKey(nameof(TrainerId))]
        public User Trainer { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }
        public ICollection<Enrollment> Enrollments { get; set; } = new List<Enrollment>();
    }
}

File: demos_back_golang/internal/models/class.go
package models

type Class struct {
	ID        int64  `json:"id"`
	DateTime  string `json:"datetime"`
	Location  string `json:"location"`
	Price     int64  `json:"price"`
	TrainerID int64  `json:"trainer_id"`
	CreatedAt string `json:"created_at"`
	UpdatedAT string `json:"updated_at"`
}

type CreateClassRequest struct {
	DateTime string `json:"datetime"`
	Location string `json:"location"`
	Price    int64  `json:"price"`
}

type UpdateClassRequest struct {
	DateTime  string `json:"datetime"`
	Location  string `json:"location"`
	Price     int64  `json:"price"`
	TrainerID int64  `json:"trainer_id"`
}

type Trainer struct {
	ID       int64  `json:"id"`
	Username string `json:"username"`
}

type ClassResponse struct {
	ID       int64   `json:"id"`
	DateTime string  `json:"datetime"`
	Location string  `json:"location"`
	Price    int64   `json:"price"`
	Trainer  Trainer `json:"trainer"`
}
